core/application/impl/chain_spec_impl.cpp

```diff
@@ -189,9 +212,16 @@ namespace kagome::application {
 
     auto code_substitutes_opt = tree.get_child_optional("codeSubstitutes");
     if (code_substitutes_opt.has_value()) {
-      for (const auto &[_hash, _code] : code_substitutes_opt.value()) {
-        OUTCOME_TRY(hash_processed, common::Hash256::fromHexWithPrefix(_hash));
-        if (hash_processed == hash) {
+      for (const auto &[_block_id, _code] : code_substitutes_opt.value()) {
+        OUTCOME_TRY(block_id_parsed, parseBlockId(_block_id));
+        if (visit_in_place(
```

Comment:
Perhaps it can be replaced with std::holds_alternative, since BlockId is a variant? Or define helper methods in the block id itself.

Reply:
Does it work for boost::variant?